sandbox/lua/encoders/schema_influx_write.lua

```diff
@@ -129,12 +130,17 @@ require "string"
 local interp = require "msg_interpolate"
 
 local name_prefix_orig  = read_config("name_prefix") or "name"
-local name_prefix = name_prefix_orig
 local use_subs
-if string.find(name_prefix, "%%{[%w%p]-}") then
+if string.find(name_prefix_orig, "%%{[%w%p]-}") then
     use_subs = true
 end
 
+if name_prefix_orig == "**none**" then
+    name_prefix_orig = ""
+else
+    name_prefix_orig = name_prefix_orig.."."
+end
+
```

Comment:
The two changes above can be replaced by:

```lua
local use_subs
if name_prefix_orig != "" then
    if string.find(name_prefix_orig, "%%{[%w%p]-}") then
        use_subs = true
    end
    name_prefix_orig = name_prefix_orig.."."
end
```

Then the rest of the code below should work as desired.